### PR TITLE
MIGENG-148: MA - Reporting - Analysis (Report) List Filtering

### PR DESCRIPTION
**Pull request by carlosthe19916**
add filterText QueryParam to `/report` to allow filtering reports/analysis. I took the chance of removing `summary` query param too.

https://jira.coreos.com/browse/MIGENG-148

### CODE REVIEW

**Comment by mrizzi** [comment]
Waiting for https://jira.coreos.com/browse/MIGENG-155 to be implemented before reviewing this PR since we need a real workflow to test this.

File: src/test/java/org/jboss/xavier/integrations/route/XmlRoutes_RestReportTest.java [hunk 1]
```diff
@@ -63,38 +63,62 @@ public void setup() {
...
         camelContext.start();
         camelContext.startRoute("reports-get-all");
-        Map<String, String> variables = new HashMap<>();
-        variables.put("summary", "true");
-        restTemplate.getForEntity(camel_context + "report?summary={summary}", String.class, variables);
+        restTemplate.getForEntity(camel_context + "report", String.class);
 
         //Then
-        verify(analysisService).findReports(0, 10);
+        verify(analysisService).findReports(anyInt(), anyInt());
```
Comment (mrizzi) [inline src/test/java/org/jboss/xavier/integrations/route/XmlRoutes_RestReportTest.java]: I think we should check that default values (`page="0"` and `size="10"`) for the params are applied in this test

**Review by mrizzi — COMMENTED** [review]
It works fine: i've just added a comment for a check in one test.

File: src/test/java/org/jboss/xavier/integrations/route/XmlRoutes_RestReportTest.java (hunk 1 above)
Comment (carlosthe19916) [reply src/test/java/org/jboss/xavier/integrations/route/XmlRoutes_RestReportTest.java]: My last commit changed anyInt() by static pagination values

**Review by mrizzi — APPROVED** [review]
Thanks @carlosthe19916: it looks good to me :+1:

**Pull request merged**